src/bun.js/bindings/JSBufferList.cpp

```diff
@@ -170,19 +192,23 @@ JSC::JSValue JSBufferList::_getBuffer(JSC::VM& vm, JSC::JSGlobalObject* lexicalG
     if (UNLIKELY(!array)) {
         return throwTypeError(lexicalGlobalObject, throwScope, "_getBuffer can only be called when all buffers are Uint8Array"_s);
     }
-    const size_t len = array->byteLength();
+    const size_t len = array->byteLength() - m_read;
     size_t n = total;
 
     if (n == len) {
         m_deque.removeFirst();
+        if (m_read > 0) {
+            auto buffer = array->possiblySharedBuffer();
+            array = JSC::JSUint8Array::create(lexicalGlobalObject, subclassStructure, buffer, m_read, len);
```

Comment:
This needs a `RETURN_IF_EXCEPTION`. If the array is detached, it will throw

Reply:
Done in https://github.com/oven-sh/bun/pull/1735/commits/e56d73e1d58cdb981d76d7f522bb02b49e2a33c1